lib/stages/mergeBeamBuffer.cpp

```diff
@@ -0,0 +1,130 @@
+#include <functional>  // for _Bind_helper<>::type, bind, function
+#include <regex>       // for match_results<>::_Base_type
+#include <stdexcept>   // for runtime_error
+#include <string>      // for string
+#include <visUtil.hpp> // for frameID, modulo
+#include <iostream>    // for cout
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::Stage;
+using std::string;
+
+REGISTER_KOTEKAN_STAGE(mergeBeamBuffer);
+
+mergeBeamBuffer::mergeBeamBuffer(Config& config_, const std::string& unique_name,
+                               bufferContainer& buffer_container) :
+    Stage(config_, unique_name, buffer_container,
+          std::bind(&mergeBeamBuffer::main_thread, this)) {
+
+    // Apply config
+    _samples_per_data_set = config.get<uint32_t>(unique_name, "samples_per_data_set");
+    _sub_frames_per_merged_frame = config.get<uint32_t>(unique_name, "sub_frames_per_merged_frame"); 
+    in_buf = get_buffer("in_buf");
+    register_consumer(in_buf, unique_name.c_str());
+
+    out_buf = get_buffer("out_buf");
+
+    register_producer(out_buf, unique_name.c_str());
+    
+}
+
+mergeBeamBuffer::~mergeBeamBuffer() {}
+
+
+void mergeBeamBuffer::main_thread() {
+    frameID in_buffer_ID(in_buf);
+    uint8_t* in_frame = nullptr;
+    frameID out_buffer_ID(out_buf);
+    uint8_t* out_frame = nullptr;
+    // Since the input buffer is a ring buffer and the in_buf_ID goes back to zero
+    // when the ID reaches the num_frame, we need something something to keep track
+    // on the incoming frame number
+    uint32_t in_frame_counter = 0;
+    
+    while (!stop_thread) {
+        // Get an input buffer, This call is blocking!
+        in_frame = wait_for_full_frame(in_buf, unique_name.c_str(), in_buffer_ID);
+        if (in_frame == nullptr)
+            break;
+
+	BeamMetadata* in_metadata = (BeamMetadata*)get_metadata(in_buf, in_buffer_ID);
+        
+        // Compute the frame index in the outbuffer
+        uint32_t sub_frame_index = in_frame_counter % _sub_frames_per_merged_frame;
+        // compute the size of the sub_frame of the out_frame
+        uint32_t FreqIDBeamMeta_size = sizeof(FreqIDBeamMetadata);
+        uint32_t sub_frame_size = FreqIDBeamMeta_size + in_buf -> frame_size;
+        
+	/// Get a new merged frame for a new set of sub frames. 
+	if (sub_frame_index == 0){
+            out_frame = (uint8_t*)wait_for_empty_frame(out_buf, unique_name.c_str(), out_buffer_ID);
+            if (out_frame == nullptr)
+                break;
+	    allocate_new_metadata_object(out_buf, out_buffer_ID);
+            MergedBeamMetadata* out_metadata = (MergedBeamMetadata*)get_metadata(out_buf, out_buffer_ID);
+	    // populate the merged_buffer/out_buffer frame header.
+	    out_metadata -> sub_frame_pre_frame = _sub_frames_per_merged_frame;
+            out_metadata -> sub_frame_metadata_size = FreqIDBeamMeta_size;
+            out_metadata -> sub_frame_data_size = in_buf -> frame_size;
```

Comment:
Our coding standard doesn't use spaces here.
```suggestion
	    out_metadata->sub_frame_pre_frame = _sub_frames_per_merged_frame;
            out_metadata->sub_frame_metadata_size = FreqIDBeamMeta_size;
            out_metadata->sub_frame_data_size = in_buf->frame_size;
```